iaso/tasks/process_mobile_bulk_upload.py

```diff
@@ -0,0 +1,163 @@
+                    with zip_ref.open(ORG_UNITS_JSON) as file:
+                        new_org_units = import_org_units(json.load(file), user, project.app_id)
+                        stats["new_org_units"] = len(new_org_units)
+                else:
+                    logger.info(f"The file {ORG_UNITS_JSON} does not exist in the zip file.")
+
+                if INSTANCES_JSON in zip_ref.namelist():
+                    with zip_ref.open(INSTANCES_JSON) as file:
+                        import_instances(json.load(file), user, project.app_id)
+                else:
+                    logger.info(f"The file {INSTANCES_JSON} does not exist in the zip file.")
+
+                logger.info("Processing forms and files")
+                new_instance_files = []
+                for directory in zipfile.Path(zip_ref).iterdir():
+                    if not directory.is_dir():
+                        continue
+
+                    instance = process_instance_xml(directory, zip_ref, user)
+                    stats["new_instances"] += 1
+                    new_instance_files += process_instance_files(directory, instance)
+
+                duplicated_count = duplicate_instance_files(new_instance_files)
+                stats["new_instance_files"] = len(new_instance_files) + duplicated_count
+
+    except Exception as e:
+        logger.error("Exception! Rolling back import: " + str(e))
+        api_import.has_problem = True
+        api_import.exception = format_exc()
+        api_import.save()
+        raise e
+
+    the_task.report_success_with_result(
+        message=result_message(user, project, start_date, start_time, stats),
+    )
+
+
+def process_instance_xml(directory, zip_ref, user):
+    instance = Instance.objects.get(uuid=directory.name)
+    logger.info(f"Processing instance {instance.uuid}")
+    with zip_ref.open(os.path.join(directory.name, instance.file_name), "r") as f:
+        instance.file = File(f)
+        instance.created_by = user
+        instance.last_modified_by = user
+        instance.save()
+        instance.get_and_save_json_of_xml()
+        try:
+            instance.convert_location_from_field()
+            instance.convert_device()
+            instance.convert_correlation()
+        except ValueError as error:
+            logger.exception(error)
+
+    return instance
+
+
+def process_instance_files(directory, instance):
+    instance_files = []
+    for instance_file in directory.iterdir():
+        if instance_file.name != instance.file_name:
+            with instance_file.open("rb") as f:
+                logger.info(f"\tProcessing attachment {instance_file.name}")
+                fi = InstanceFile()
+                fi.file = File(f)
+                fi.instance_id = instance.id
+                fi.name = instance_file.name
+                fi.save()
+                instance_files.append(fi)
+
+    return instance_files
+
+
+# Trypelim-specific
```

Comment:
Main issue I see for merging this is how this code is Trypelim specific (although it won't affect any form that doesn't have a `serie_id` attribute)